cmd/dep/ensure.go

```diff
@@ -145,17 +210,11 @@ func (cmd *ensureCommand) Run(ctx *dep.Ctx, args []string) error {
 			Constraint: pc.Constraint,
 		}
 
-		if p.Lock != nil {
-			for i, lp := range p.Lock.P {
-				if lp.Ident() == pc.Ident {
-					p.Lock.P = append(p.Lock.P[:i], p.Lock.P[i+1:]...)
-					break
-				}
-			}
-		}
+		// Ignore the lockfile for this dependency and allow its version to change
+		params.ToChange = append(params.ToChange, pc.Ident.ProjectRoot)
```

Comment:
Sure thing, I'll go back to what it was doing before.

Would you help me understand what the original code was doing? I had originally thought that it was tweaking the in-memory representation of the lock to allow the package version to change, and that switching to `ToChange` was equivalent. Sounds like I am misunderstanding that however. 😊